source/mir/random/discrete.d

```diff
@@ -135,9 +137,10 @@ struct Discrete(T)
             arr[sm.index] = AltPair(sm.prob, lg.index);
 
             // update larger & reinsert
-            lg.prob += sm.prob - 1;
+            lg.prob += sm.prob;
+            lg.prob -= 1;
 
-            if (lg.prob < 1)
+            if (lg.prob < 1 || lg.prob.approxEqual(1, 1e-13, 1e-13))
```

Comment:
> we will have fp pragmas
> ... but what should I do for now? Just enable the sampling test for x86_64 and double?

I think it will take 2-6 months until we get the pragma, so what should I do for now?